### PR TITLE
Add cilium network policies support

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/23014
Towards https://github.com/giantswarm/roadmap/issues/1949

Otherwise:

```
kube-apiserver-oa92854gla2fac kube-apiserver W0629 16:50:42.683089       1 dispatcher.go:195] Failed calling webhook, failing closed mutate-policy.kyverno.svc: failed calling webhook "mutate-policy.kyverno.svc": failed to call webhook: Post "https://kyverno-svc.kyverno.svc:443/policymutate?timeout=10s": dial tcp 172.31.207.106:443: connect: operation not permitted
```

### CODE REVIEW

**Comment by stone-z** [comment]
Can the cilium policy instead live under [`/templates`](https://github.com/giantswarm/kyverno-app/tree/main/helm/kyverno/templates) and an associated values key (same as PSPs, VPA, etc.)?
Everything under `/charts` is pure upstream and we have automation to keep it in sync

**Comment by kopiczko** [comment]
> Can the cilium policy instead live under [`/templates`](https://github.com/giantswarm/kyverno-app/tree/main/helm/kyverno/templates) and an associated values key (same as PSPs, VPA, etc.)? Everything under `/charts` is pure upstream and we have automation to keep it in sync

Sure I will do that

**Comment by kopiczko** [comment]
@stone-z I hope it's what you want now

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add cilium network policies support (enabled with `kyverno.ciliumNetworkPolicy.enabled=true`).
```
Comment (stone-z) [inline CHANGELOG.md:12]:
```suggestion
- Add cilium network policies support (enabled with `ciliumNetworkPolicy.enabled=true`).
```

**Review by stone-z — APPROVED** [review]
Correction in the changelog but otherwise LGTMIIWFY :rocket:

**Review by fhielpos — APPROVED** [review]
LGTM 🚀

**Pull request merged**